forte/mrdsrg-spin-integrated/dsrg_mrpt2_grad/dsrg_mrpt2_deriv_tensors.cc

```diff
@@ -0,0 +1,515 @@
+    dlamb3_abb("KxYZuVW") += cc1b_r("KZV") * Gamma1_.block("aa")("xu") * Gamma1_.block("AA")("YW");
+    dlamb3_abb("KxYZuVW") += cc1b_n("KYW") * Gamma1_.block("AA")("ZV") * Gamma1_.block("aa")("xu");
+    dlamb3_abb("KxYZuVW") += cc1b_r("KYW") * Gamma1_.block("AA")("ZV") * Gamma1_.block("aa")("xu");
+}
+
+/**
+ * Initializing the two-body density Gamma2_.
+ *
+ * NOTICE: this function shall be deprecated in the future.
+ */
+void DSRG_MRPT2::set_density() {
+    Gamma2_ = BTF_->build(CoreTensor, "Gamma2_", spin_cases({"aaaa"}));
+
+    Gamma2_.block("aaaa")("pqrs") = rdms_.g2aa()("pqrs");
+    Gamma2_.block("aAaA")("pqrs") = rdms_.g2ab()("pqrs");
+    Gamma2_.block("AAAA")("pqrs") = rdms_.g2bb()("pqrs");
+}
+
+/**
+ * Initializing the one-electron integral H.
+ *
+ * NOTICE: this function shall be deprecated in the future.
+ */
+void DSRG_MRPT2::set_h() {
+    H = BTF_->build(CoreTensor, "One-Electron Integral", spin_cases({"gg"}));
+    H.iterate([&](const std::vector<size_t>& i, const std::vector<SpinType>& spin, double& value) {
+        if (spin[0] == AlphaSpin) {
+            value = ints_->oei_a(i[0], i[1]);
+        } else {
+            value = ints_->oei_b(i[0], i[1]);
+        }
+    });
+}
+
+/**
+ * Initializing the ERIs V.
+ *
+ * NOTICE: this function shall be deprecated in the future.
+ */
+void DSRG_MRPT2::set_v() {
+    V = BTF_->build(CoreTensor, "Electron Repulsion Integral",
+                    spin_cases({"gphh", "pghh", "ppgh", "pphg", "gchc", "pghc", "pcgc", "pchg",
+                                "gcpc", "hgpc", "hcgc", "hcpg", "gccc", "cgcc", "ccgc", "cccg",
+                                "gcvc", "vgvc", "vcgc", "vcvg", "cgch", "gpch", "cpcg", "cpgh",
+                                "cgcp", "ghcp", "chcg", "chgp", "cgcv", "gvcv", "cvcg", "cvgv"}));
+
+    V.iterate([&](const std::vector<size_t>& i, const std::vector<SpinType>& spin, double& value) {
+        if (spin[0] == AlphaSpin) {
+            if (spin[1] == AlphaSpin) {
+                value = ints_->aptei_aa(i[0], i[1], i[2], i[3]);
+            } else {
+                value = ints_->aptei_ab(i[0], i[1], i[2], i[3]);
+            }
+        } else if (spin[1] == BetaSpin) {
+            value = ints_->aptei_bb(i[0], i[1], i[2], i[3]);
+        }
+    });
+
+    V_N_Alpha = BTF_->build(CoreTensor,
+                            "normal Dimention-reduced Electron Repulsion Integral alpha", {"gg"});
+    V_N_Beta = BTF_->build(CoreTensor, "normal Dimention-reduced Electron Repulsion Integral beta",
+                           {"gg"});
+    V_R_Beta = BTF_->build(
+        CoreTensor, "index-reversed Dimention-reduced Electron Repulsion Integral beta", {"GG"});
+    V_all_Beta = BTF_->build(
+        CoreTensor, "normal Dimention-reduced Electron Repulsion Integral all beta", {"GG"});
+
+    // Summation of V["pmqm"] over index "m" or V["mpmq"] over index "m"
+    V_N_Alpha["pq"] = V["pmqn"] * I["mn"];
```

Comment:
"Normal" and "Reversed" is not standard terminology. Do you mean Coulomb and Exchange?